frontend/widgets/OBSBasic_Transitions.cpp

```diff
@@ -496,7 +518,10 @@ void OBSBasic::on_transitionRemove_clicked()
 		}
 	}
 
-	ui->transitions->removeItem(idx);
+	transitionUuids.erase(std::find(transitionUuids.begin(), transitionUuids.end(), currentTransitionUuid));
+	transitionNameToUuids.erase(obs_source_get_name(tr));
```

Comment:
If I understood the C++ docs correctly, `erase` can take an argument of the key type and will only throw if the comparator itself throws (so providing an `std::string` key for a non-existent element should just do nothing).

If it is preferred to explicitly use an iterator, I'd rather have that on its own line to make the code more readable, but the shortened form should be sufficient.

I'm more concerned about the second line, as it relies on implicit conversion between `char *` and `std::string`. I dunno if that conversion has defined behaviour when a `NULL` pointer is provided - if it does the code should be safe (as the empty string object will just fail to match anything in the collection).

Reply:
Second line refactored.

Reply:
For `transitionUuids`, https://en.cppreference.com/w/cpp/container/vector/erase only list iterator based implementation (ignoring C++ 20 stuff).